Index: Source/Modules/csharp.cxx
===================================================================
RCS file: /cvsroot/swig/SWIG/Source/Modules/csharp.cxx,v
retrieving revision 1.94
diff -u -r1.94 csharp.cxx
--- Source/Modules/csharp.cxx	16 Oct 2006 20:09:21 -0000	1.94
+++ Source/Modules/csharp.cxx	20 Oct 2006 12:20:21 -0000
@@ -1916,12 +1916,16 @@
     if (csattributes)
       Printf(function_code, "  %s\n", csattributes);
     const String *methodmods = Getattr(n,"feature:cs:methodmodifiers");
-    if (methodmods) {
+    if (methodmods && !is_smart_pointer()) {
       Printf(function_code, "  %s ", methodmods);
     } else {
-      methodmods = (!is_public(n) ? protected_string : public_string);
+      if (methodmods)
+	/* Method of smart pointers should not be defined "override"  */
+	methodmods = Strcmp(methodmods, public_string) ? protected_string : public_string;
+      else
+	methodmods = (!is_public(n) ? protected_string : public_string);
       Printf(function_code, "  %s ", methodmods);
-      if (Getattr(n,"override"))
+      if (Getattr(n,"override") && !is_smart_pointer())
 	Printf(function_code, "override ");
       else if (checkAttribute(n, "storage", "virtual"))
 	Printf(function_code, "virtual ");
